src/librustc_privacy/lib.rs

```diff
@@ -412,14 +412,27 @@ impl<'b, 'a, 'tcx> ReachEverythingInTheInterfaceVisitor<'b, 'a, 'tcx> {
         self.ev.tcx.impl_trait_ref(self.item_def_id).visit_with(self);
         self
     }
+
+    fn visit_trait_ref(&mut self, trait_ref: ty::TraitRef<'tcx>) -> bool {
```

Comment:
You missed the `impl_trait_ref` caller for this method! Although since there is no other user, you could maybe move the whole method into there.